src/OmniSharp.Abstractions/Endpoints/IGotoDefinition.cs

```diff
@@ -0,0 +1,11 @@
+using System.Threading.Tasks;
+using OmniSharp.Models;
+
+namespace OmniSharp.Endpoints
+{
+    public interface IGotoDefintion
+    {
+        Task<bool> IsApplicableTo(GotoDefinitionRequest request);
```

Comment:
The one issue I see with the delegate route is that we can't initialize any properties from the MEF host.  So if we need any external services, they will all have to be done on demand in each delegate.

Also the workspace will have to stay static, or become part of the delegate itself, so we can hand it.